cmake/std/atdm/cee-rhel6/environment.sh

```diff
@@ -88,6 +88,19 @@ elif [[ "$ATDM_CONFIG_COMPILER" == "CLANG-5.0.1_OPENMPI-4.0.2" ]]; then
   fi
   export ATDM_CONFIG_MKL_ROOT=${CBLAS_ROOT}
 
+elif [[ "$ATDM_CONFIG_COMPILER" == "CLANG-9.0.1_OPENMPI-4.0.2" ]]; then
+  module load sparc-dev/clang-9.0.1_openmpi-4.0.2
+  export OMPI_CXX=`which clang++`
+  export OMPI_CC=`which clang`
+  export OMPI_FC=`which gfortran`
+  export MPICC=`which mpicc`
+  export MPICXX=`which mpicxx`
+  export MPIF90=`which mpif90`
+  if [[ "$ATDM_CONFIG_ENABLE_STRONG_WARNINGS" == "1" ]]; then
+    export ATDM_CONFIG_CXX_FLAGS="${ATDM_CONFIG_GNU_CXX_WARNINGS}"
+  fi
+  export ATDM_CONFIG_MKL_ROOT=${CBLAS_ROOT}
```

Comment:
Suggestion: Move this and https://github.com/trilinos/Trilinos/pull/7204/files#diff-5ba41a58f56d71bbeb2b044fc5467e6bR80-R89 outside of the if blocks so that the if blocks just contain the `module load sparc-dev/...`

Reply:
No, if you look at the way this is defined for intel-18 it is different:

```
export ATDM_CONFIG_MKL_ROOT=/sierra/sntools/SDK/compilers/intel/composer_xe_2018.5.274/compilers_and_libraries/linux/
```

Reply:
However, given how common `export ATDM_CONFIG_MKL_ROOT=${CBLAS_ROOT}` is, we set it to `unset ATDM_CONFIG_MKL_ROOT` before the if/elif block and if it is not set by the intel-18 block, then sets it as `export ATDM_CONFIG_MKL_ROOT=${CBLAS_ROOT}`.  That will remove the duplication.  But then we will have to test every build configuration (testing at least Teuchos which uses BLAS and LAPACK TPLs) to ensure we have not broken anything and check the result carefully before we mere.

Reply:
@bartlettroscoe: For all 3 branches, the entire block:
```bash
  export OMPI_CXX=`which clang++`
  export OMPI_CC=`which clang`
  export OMPI_FC=`which gfortran`
  export MPICC=`which mpicc`
  export MPICXX=`which mpicxx`
  export MPIF90=`which mpif90`
  if [[ "$ATDM_CONFIG_ENABLE_STRONG_WARNINGS" == "1" ]]; then
    export ATDM_CONFIG_CXX_FLAGS="${ATDM_CONFIG_GNU_CXX_WARNINGS}"
  fi
  export ATDM_CONFIG_MKL_ROOT=${CBLAS_ROOT}
```

should be moved below the if-elif statement. CBLAS_ROOT will take care of any differences in paths.

Reply:
@e10harvey, can you post a PR against this PR branch with that suggested change?  You can branch off of this branch and the post a PR against the branch `bartlettroscoe:atdv-322-cee-rhel6-clang-9.0.1` in my GitHub fork.  I think that avoids blurring the line against "author" and "reviewer".

Reply:
@bartlettroscoe: Does [this](https://github.com/trilinos/Trilinos/pull/7204#discussion_r411458675) make sense?

Reply:
> Does this make sense?

@e10harvey, what about the module loads?

Reply:
The module loads should stay in their respective conditional bodies.

Reply:
> The module loads should stay in their respective conditional bodies.

You mean for just the clang configurations?

Reply:
Yes. For https://github.com/trilinos/Trilinos/pull/7204/files#diff-5ba41a58f56d71bbeb2b044fc5467e6bR65-R102.

Reply:
@e10harvey, the duplicated code is gone in the most recent branch.